[Pawandeep Singh] iP

pss-coder opened the pull request:
# Gnosis 
> The spark to meet your needs

**Things to note:** 
1. Gnosis is a Command-Line interface application
2. Pre-requisites: Java 11
3. To test it -> download the Jar file [`v0.1 here`](https://github.com/pss-coder/ip/releases/download/A-Jar/gnosis.jar)

Gnosis provides user the following features: 

- Add 3 kinds of tasks (_todos, deadlines, events_)
- Search and filter your tasks 
- All stored locally in a CSV File

Features coming soon: 
Week 4: 
- [ ]  Gradle implementation 
- [ ]  GUI via JavaFX 


Gnosis ```main``` program code ⭐ : 

```Java
public static void main(String[] args) {
    // initialise gnosis program
    Gnosis gnosis = new Gnosis();

    // runs program and waits for user input
    Scanner sc = new Scanner(System.in);
    gnosis.run(sc);

    sc.close();
}

```

Commit:
Add Gradle support

Commit:
Initial Commit

Commit:
Create level-1 Gnosis

Commit:
Update Gnosis to level-2

Commit:
Update Gnosis to level-3

Commit:
Update Gnosis to level-4

Commit:
Automate Text UI Texting

Commit:
Update Gnosis to Level-5

Commit:
Update Gnosis to Level-6

Commit:
Create GnosisCommand Enum

Commit:
Update Documentation

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update Documentation

pss-coder committed:
Update Documentation

pss-coder committed:
Update Documentation

Commit:
Create FileManager.java

Commit:
Refactor Task Class

Commit:
Update FileManager.java read/write to Task.java

Commit:
Provide Save/Load task feature to Gnosis

Commit:
Update DateTime formatting for Gnosis

Commit:
Fix file merge conflicts with Level-7 and Level-8

Commit:
Refactor codebase to follow MVC Architecture

Commit:
Refactor codebase file package

pss-coder committed:
Update README.md

pss-coder committed:
Update README.md

pss-coder committed:
Update README.md

Commit:
Rename Task package to task

pss-coder committed:
Create JUnit testing for gnosis classes

pss-coder committed:
Merge remote-tracking branch 'origin/master'

pss-coder committed:
Implement FIND command to gnosis

pss-coder committed:
Update packages to coding convention

pss-coder committed:
Update codebase to meet code convention

pss-coder committed:
Create JavaDoc for gnosis and model packages

pss-coder committed:
Merge branch 'branch-A-CodingStandard'

pss-coder committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/gnosis/Gnosis.java
#	src/main/java/gnosis/GnosisController.java
#	src/main/java/gnosis/model/Command.java
#	src/main/java/gnosis/model/Deadline.java
#	src/main/java/gnosis/model/Event.java
#	src/main/java/gnosis/model/Task.java
#	src/main/java/gnosis/model/Todo.java
#	src/main/java/gnosis/ui/GnosisUI.java
#	src/main/java/gnosis/util/GnosisException.java
#	src/main/java/model/TaskType.java

pss-coder committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/gnosis/GnosisUI.java
#	src/main/java/gnosis/controller/GnosisController.java

pss-coder committed:
Add documentation to Find Task Command

pss-coder committed:
Fix merge conflict in GnosisController

pss-coder committed:
Merge branch 'add-gradle-support' of https://github.com/pss-coder/ip into branch-A-Gradle

pss-coder committed:
Automate Gnosis Project with Gradle

pss-coder committed:
Merge branch 'branch-A-Gradle'

pss-coder committed:
Fix UI Display message

pss-coder committed:
Use checkstyle with Gradle

pss-coder committed:
Merge branch 'branch-A-CheckStyle'

pss-coder committed:
Set up Gnosis with JavaFX

pss-coder committed:
Merge branch 'branch-Level-10'

pss-coder committed:
Improve GnosisUI Codebase using Varargs

pss-coder committed:
Merge branch 'branch-A-Varargs'

cedricongjh commented on src/main/java/gnosis/controller/GnosisController.java:
```diff
@@ -0,0 +1,119 @@
+     * @throws NumberFormatException if taskInput for DONE and DELETE command cannot be converted to integer.
+     */
+    public void executeTaskCommand(Command gc, String taskInput) throws GnosisException, NumberFormatException {
+        switch (gc) {
```
According to the coding standard, switch statements should have no indentation for the cases:

```java
switch(a) {
case "1":
case "2":
}

```

pss-coder replied:
Thanks for the lookout, I think I might have lost the convention when I merged the branches at some point. Thanks !

cedricongjh commented on src/main/java/gnosis/ui/GnosisUI.java:
```diff
@@ -0,0 +1,180 @@
+ *
+ * @author Pawandeep Singh
+ * */
+public class GnosisUI extends AnchorPane {
```
Class names should be in camelCase, so perhaps this could be changed to `GnosisUi`

cedricongjh reviewed (commented):
Besides the 2 that I commented on, good following of the coding standard!

muhammad-khair commented on src/main/java/gnosis/task/TaskCommandManager.java:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    // helper function
+    public static LocalDateTime StringToDate(String dateString) throws GnosisException {
```
Method name should not start with a capital letter, i.e. should be `stringToDate`. Also, may I ask why is your method a `public static` method, if you place a comment above it saying it's a helper function? If it is a helper function and is called in the class, maybe it should just be a `private` method?

muhammad-khair commented on src/main/java/gnosis/task/TaskStorageManager.java:
```diff
@@ -0,0 +1,104 @@
+            tasks = Files.newBufferedReader(Paths.get(FILE_PATH))
+                    .lines()
+                    .skip(1)
+                    .map(s -> {
```
Any possible reason why your `map` method has a nested function block in it? Maybe to make your code neater, you can try and push the function block out as a private method, if its not state-dependent. Example:
``` java
public static List<Task> getTasksFromFile() {
    List<Task> tasks = new ArrayList<>();
    try {
        tasks = Files.newBufferedReader(Paths.get(FILE_PATH))
                .lines()
                .skip(1)
                .map(this::parseTask)
                .collect(Collectors.toList());
    } catch (IOException e) {
        e.printStackTrace();
    }
    return tasks;
}

private Task parseTask(String s) throws IOException {
    // migrated method contents omitted
}
```

muhammad-khair commented on src/main/java/gnosis/util/GnosisConstants.java:
```diff
@@ -0,0 +1,58 @@
+ * @version Level-6
+ *
+ * */
+public class GnosisConstants {
```
I really like the neat storage of messages here :)

muhammad-khair reviewed (commented):
Overall, I really like the neat code blocks and not so lengthy methods in each of them. Keep up the neat structure and ensure you format your code according to the standards as mentioned by the previous PR comments :)

meerian reviewed (commented):
Seems good, no violation of coding standards seen! Keep up the good work! 👍

pss-coder committed:
Improve Codebase styling convention

pss-coder committed:
Update Gnosis Ui to GridPane

pss-coder committed:
Improve Gnosis UI

pss-coder committed:
Update GnosisMainWindow UI

pss-coder committed:
Test GnosisUI userInput assertion

pss-coder committed:
Update TaskCommandManager streams implementation

pss-coder committed:
Update TaskCommand ActionHandler

pss-coder committed:
Merge pull request #2 from pss-coder/branch-A-Assertions

Test GnosisUI userInput assertion

pss-coder committed:
Merge pull request #3 from pss-coder/branch-A-Streams

Update TaskCommandManager streams implementation

pss-coder committed:
Merge branch 'master' into branch-A-CodeQuality

pss-coder committed:
Merge pull request #1 from pss-coder/branch-A-CodeQuality

Update TaskCommand ActionHandler

pss-coder committed:
Create gradle.yml

Project has no Continuous Integration set up.

Setting Continuous Integration will automate project worflow.

Let's set up Continuous Integration using Github Actions.

Github Actions makes it easy to automate project workflow.

Refer to more info about Github Action here:
https://github.com/features/actions

pss-coder committed:
Update gradle.yml

pss-coder committed:
Remove gradle I/O redirection automated tests

pss-coder committed:
Improve codebase checkstyle

pss-coder committed:
Merge remote-tracking branch 'origin/master'

pss-coder committed:
Improve CodeStyle

pss-coder committed:
Remove isUIListening CLI test case

pss-coder committed:
Update Gnosis test cases

pss-coder committed:
Update Gnosis test cases

pss-coder committed:
Implement Category D-Places

Gnosis only supports Task management.

Gnosis should also support Place management to increase functionality.

Let's begin a Places and Visited command to support Place management.

Using Places and Visited will be a good start to slowly build our place management functionality.

pss-coder committed:
Refactor codebase

pss-coder committed:
Update Export Feature

pss-coder committed:
Improve Gnosis UI

pss-coder committed:
Update UI Codebase

pss-coder committed:
Update Codestyle

pss-coder committed:
Update codestyle

pss-coder committed:
Update ReadME documentation

pss-coder committed:
Update Testcases

pss-coder committed:
Update Documentation

pss-coder committed:
Improve codebase javadocs

Commit:
Update README.md

pss-coder commented:
## Gnosis v0.2 is ready
click [here](https://github.com/pss-coder/ip/releases/download/A-Release/Gnosis.jar) to download

Commit:
Update README.md

Commit:
Update README.md
